backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/user/role/repository/UserRoleRepository.java

```diff
@@ -118,4 +118,16 @@ private UserRole toUserRole(@NonNull final UserAssignedRoleId assignedRoleId, @N
 				.userAssignedRoleId(assignedRoleId)
 				.build();
 	}
+
+	public List<UserId> getAssignedUsers(@NonNull final UserRoleId userRoleId)
+	{
+		return queryBL.createQueryBuilder(I_C_User_Assigned_Role.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_C_User_Assigned_Role.COLUMNNAME_C_User_Role_ID, userRoleId)
+				.create()
+				.stream()
+				.map(I_C_User_Assigned_Role::getAD_User_ID)
+				.map(UserId::ofRepoId)
+				.collect(ImmutableList.toImmutableList());
```

Comment:
```suggestion
				.listDistinct(I_C_User_Assigned_Role.COLUMNNAME_C_User_Role_ID, Integer.class)
				.stream()
				.map(UserId::ofRepoId)
				.collect(ImmutableList.toImmutableList());
```

we can avoid loading them